### PR TITLE
[PROGRAMMES-6458] Update composer Dependencies

### PR DESCRIPTION
**Pull request by benolot**
Ticket: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6458

Okay so this is horrible fixes to something that shouldn't have been an issue.

Basically PHPStan now throws errors because we don't account for the fact that something like json_encode can return false on error or mb_strlen returning false if the encoding is incorrect. This isn't the best fix for these issues, but they now pass all tests and nothing appears broken so far.

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by benolot** [comment]
Test this please

File: src/BlogsService/Domain/ContentBlock/Code.php [hunk 1]
```diff
@@ -19,7 +19,8 @@ public function __construct(string $code)
     public function getCharacterCount(): int
     {
         if ($this->charCount === null) {
-            $this->charCount = mb_strlen($this->code);
+            $this->charCount = mb_strlen($this->code) ?: 0;
+//            $this->charCount = $this->charCount ? $this->charCount : 0;
```
Comment (votemike) [inline src/BlogsService/Domain/ContentBlock/Code.php]: What is this?

File: src/BlogsService/Domain/ContentBlock/Prose.php [hunk 2]
```diff
@@ -19,7 +19,8 @@ public function __construct(string $prose)
     public function getCharacterCount(): int
     {
         if ($this->charCount === null) {
-            $this->charCount = mb_strlen(strip_tags($this->prose));
+            $this->charCount = mb_strlen(strip_tags($this->prose)) ?: 0;
+//            $this->charCount = $this->charCount ? $this->charCount : 0;
```
Comment (votemike) [inline src/BlogsService/Domain/ContentBlock/Prose.php]: What is this?

File: src/Ds/Post/ContentBlock/ClipsBlock/ClipsBlockPresenter.php [hunk 3]
```diff
@@ -72,7 +72,7 @@ public function getPlayer(): string
                 'externalEmbedUrl' => null,
             ];
         }
-        return json_encode($player);
+        return json_encode($player)?:'';
```
Comment (votemike) [inline src/Ds/Post/ContentBlock/ClipsBlock/ClipsBlockPresenter.php]: Add some spacing please

File: src/Twig/SchemaJsonExtension.php [hunk 4]
```diff
@@ -63,11 +63,11 @@ public function generateSchemaData(): string
...
 
         if (\count($this->schemaSnippets) == 1) {
-            return json_encode(array_merge($schemaData, $this->schemaSnippets[0]));
+            return json_encode(array_merge($schemaData, $this->schemaSnippets[0]))?:'';
         }
 
         $schemaData['@graph'] = $this->schemaSnippets;
 
-        return json_encode($schemaData);
+        return json_encode($schemaData) ?: '';
```
Comment (votemike) [inline src/Twig/SchemaJsonExtension.php]: Why is this needed?

File: tests/Builders/AuthorBuilder.php [hunk 5]
```diff
@@ -1,5 +1,5 @@
 <?php
-declare(strict_types = 1);
+declare(strict_types=1);
```
Comment (votemike) [inline tests/Builders/AuthorBuilder.php]: No!
See https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6275

File: tests/FeedGenerator/FeedGeneratorTest.php [hunk 6]
```diff
@@ -118,16 +118,20 @@ private function getAtomFeed(Blog $blog, array $posts): SimpleXMLElement
     {
         $feedGenerator = $this->createFeedGenerator();
         $feedData = $feedGenerator->generateAtomFeed($blog, $posts);
+        /** @var SimpleXMLElement $result */
+        $result = simplexml_load_string($feedData);
```
Comment (votemike) [inline tests/FeedGenerator/FeedGeneratorTest.php]: Did we not ignore this error?

File: src/Twig/SchemaJsonExtension.php (hunk 4 above)
Comment (benolot) [reply src/Twig/SchemaJsonExtension.php]: json_encode can return false on failure, this throws an error in PHPStan as we discussed

File: tests/Builders/AuthorBuilder.php (hunk 5 above)
Comment (benolot) [reply tests/Builders/AuthorBuilder.php]: Thanks IntelliJ's auto indenter... Will fix

File: tests/FeedGenerator/FeedGeneratorTest.php (hunk 6 above)
Comment (benolot) [reply tests/FeedGenerator/FeedGeneratorTest.php]: No that was a different one, also todo with simpleXML

File: composer.json [hunk 7]
```diff
@@ -32,39 +32,39 @@
...
-        "bbc/gel-iconography-assets": "^1.2",
+        "bbc-rmp/translate": "^1.8.9",
+        "bbc/branding-client": "^2.5",
+        "bbc/gel-iconography-assets": "^2.1.1",
         "bbc/programmes-morph-library": "^1.0",
-        "cakephp/chronos": "^1.1",
-        "cakephp/utility": "^3.5",
-        "csa/guzzle-bundle": "^3.0",
-        "fzaninotto/faker": "^1.7",
+        "cakephp/chronos": "^1.2",
```
Comment (Tasiobg) [inline composer.json]: This shouldn't make any difference because composer update will pick the last v2 version. Not sure but I think could be better not to change it unless it actually makes a difference, lets see what the others think.
By the way this applies to the other patch and minor version changes with the exception of the ones in the v0.X because composer use some safety protection in this case 
https://getcomposer.org/doc/articles/versions.md#caret-version-range-

Comment (benolot) [reply composer.json]: Myself and Mike had this discussion on slack and we couldn't agree - Personally I think the composer.json should be kept upto date with the latest working update, to ensure it never goes back to a perhaps less secure, or less bug-fixed version, Mike preferred leaving it as it was. 
Bare in mind most of my package methods/knowledge comes from NPM before they had a lock file, so I may be entirely off the standard here.

Comment (Tasiobg) [reply composer.json]: Composer update gets the last version available in the specified range, how could increasing the minim version make any difference in the version installed?

**Comment by benolot** [comment]
Test this please

File: phpunit.xml.dist [hunk 8]
```diff
@@ -47,10 +38,5 @@
         <listener class="JohnKary\PHPUnit\Listener\SpeedTrapListener" />
     </listeners>
 
-    <logging>
```
Comment (votemike) [inline phpunit.xml.dist]: Why have you removed this?

Comment (benolot) [reply phpunit.xml.dist]: screen shot 2018-07-23 at 15 09 09
Testing! :P I'm not sure if it's actually failing, or if the warning about deprecated logging is causing the build to be marked as failing

File: composer.json (hunk 7 above)
Comment (tatitati) [reply composer.json]: To be honest is interesting this thread. In one side is true that updating composer.json to these versions is not harmful as they are just including a minimum version but these versions won't break anything. This is easy to see as they are already installed and working properly. Nothing exploded.

But in the other side is true that our concern when using composer.json is more about breaking changes. We are not too worried about an specific version when speaking about  third parties, however what we really fear is a breaking change. So this level of granularity is too low. Our concern is bigger than think  about 3.2 or 3.4. What we are really worried about is about a jump from 3.* to 4.*
This said, this allows us to be a little "beast" and simplify our use of composer just running ```composer update```.

Comment (votemike) [reply composer.json]: Not keen on updating this when it doesn't need to be
